native/src/main/java/io/ballerina/stdlib/graphql/runtime/engine/Engine.java

```diff
@@ -179,67 +188,169 @@ private static void getExecutionResult(ExecutionContext executionContext, BObjec
+            ArrayType arrayType = getArrayType(argType);
+            Type elementType = arrayType.getElementType();
+            if (listItem.getBooleanValue(VARIABLE_DEFINITION)) {
+                addListMembersFromVariableValue(listItem, valueArray, getNonNullType(elementType));
+            } else if (listItem.getIntValue(KIND_FIELD) == T_INPUT_OBJECT) {
+                BArray inputObjectFields = listItem.getArrayValue(VALUE_FIELD);
+                BMap<BString, Object> inputObjectRecord = ValueCreator.createMapValue(elementType);
+                addInputObjectTypeArgumentFromNode(inputObjectFields, inputObjectRecord, (RecordType) elementType);
+                valueArray.append(inputObjectRecord);
+            } else if (listItem.getIntValue(KIND_FIELD) == T_LIST) {
+                BArray values = listItem.getArrayValue(VALUE_FIELD);
+                BArray nestedValueArray = ValueCreator.createArrayValue(getArrayType(elementType));
+                addListTypeArgumentFromNode(values, nestedValueArray, elementType);
+                valueArray.append(nestedValueArray);
+            } else {
+                Object argValue = listItem.get(VALUE_FIELD);
+                valueArray.append(argValue);
+            }
+        }
+    }
+
+    public static void addInputObjectFieldsFromVariableValue(BObject argumentNode, BMap<BString, Object> argumentsMap,
+                                                             Type argType) {
+        BString argName = argumentNode.getStringValue(NAME_FIELD);
+        if (argType.getTag() == TypeTags.RECORD_TYPE_TAG) {
+            BMap<BString, Object> inputObjectFields = argumentNode.getMapValue(VARIABLE_VALUE_FIELD);
+            BMap<BString, Object> inputObjectRecord = ValueCreator.createMapValue(argType);
+            visitInputObjectTypeVariableValue(inputObjectFields, inputObjectRecord, (RecordType) argType);
+            argumentsMap.put(argName, inputObjectRecord);
+        } else if (argType.getTag() == TypeTags.ARRAY_TAG) {
+            BArray values = argumentNode.getArrayValue(VARIABLE_VALUE_FIELD);
+            BArray valueArray = ValueCreator.createArrayValue(getArrayType(argType));
+            visitListTypeVariableValue(values, valueArray, argType);
+            argumentsMap.put(argName, valueArray);
+        } else {
+            Object value = argumentNode.get(VARIABLE_VALUE_FIELD);
+            argumentsMap.put(argName, value);
+        }
+    }
+
+    public static void addListMembersFromVariableValue(BObject listItem, BArray valueArray, Type argType) {
+        if (argType.getTag() == TypeTags.RECORD_TYPE_TAG) {
+            BMap<BString, Object> inputObjectFields = listItem.getMapValue(VARIABLE_VALUE_FIELD);
+            BMap<BString, Object> inputObjectRecord = ValueCreator.createMapValue(argType);
+            visitInputObjectTypeVariableValue(inputObjectFields, inputObjectRecord, (RecordType) argType);
+            valueArray.append(inputObjectRecord);
```

Comment:
Why does this line not covered in tests? If a test is missing, shall we add one?

Reply:
This test is missing. Will add one.